tracer/src/Datadog.Trace/TracerManager.cs

```diff
@@ -0,0 +1,387 @@
+            DefaultServiceName = defaultServiceName;
+            LibLogSubscriber = libLogSubscriber;
+        }
+
+        /// <summary>
+        /// Gets the global <see cref="TracerManager"/> instance used by all <see cref="Tracer"/> instances
+        /// </summary>
+        public static TracerManager Instance
+        {
+            get
+            {
+                return LazyInitializer.EnsureInitialized(
+                    ref _instance,
+                    ref _globalInstanceInitialized,
+                    ref _globalInstanceLock,
+                    () => CreateInitializedTracer(settings: null, TracerManagerFactory.Instance));
+            }
+        }
+
+        /// <summary>
+        /// Gets the default service name for traces where a service name is not specified.
+        /// </summary>
+        public string DefaultServiceName { get; }
+
+        /// <summary>
+        /// Gets this tracer's settings.
+        /// </summary>
+        public TracerSettings Settings { get; }
+
+        public IAgentWriter AgentWriter { get; }
+
+        /// <summary>
+        /// Gets the tracer's scope manager, which determines which span is currently active, if any.
+        /// </summary>
+        public IScopeManager ScopeManager { get; }
+
+        /// <summary>
+        /// Gets the <see cref="ISampler"/> instance used by this <see cref="IDatadogTracer"/> instance.
+        /// </summary>
+        public ISampler Sampler { get; }
+
+        public IDogStatsd Statsd { get; }
+
+        private RuntimeMetricsWriter RuntimeMetrics { get; }
+
+        private LibLogScopeEventSubscriber LibLogSubscriber { get; }
+
+        /// <summary>
+        /// Replaces the global <see cref="TracerManager"/> settings. This affects all <see cref="Tracer"/> instances
+        /// which use the global <see cref="TracerManager"/>
+        /// </summary>
+        /// <param name="settings">The settings to use </param>
+        /// <param name="factory">The factory to use to create the <see cref="TracerManager"/></param>
+        public static void ReplaceGlobalManager(TracerSettings settings, TracerManagerFactory factory)
+        {
+            TracerManager oldManager;
+            TracerManager newManager;
+            lock (_globalInstanceLock)
+            {
+                oldManager = _instance;
+                newManager = CreateInitializedTracer(settings, factory);
+                _instance = newManager;
+                _globalInstanceInitialized = true;
+            }
+
+            if (oldManager is null)
+            {
+                return;
+            }
+
+            // Clean up the old TracerManager instance
+            oldManager._isClosing = true;
+            var agentWriterReplaced = false;
+            if (oldManager.AgentWriter != newManager.AgentWriter)
+            {
+                agentWriterReplaced = true;
+                oldManager.AgentWriter?.FlushAndCloseAsync();
```

Comment:
Hmmm, this is async... Question is whether we can safely make this `oldManager.AgentWriter?.FlushAndCloseAsync().GetAwaiter().GetResult();` or whether we have to make `ReplaceGlobalManager()` (and hence the public API) async 😕